### ISSUE TITLE
Typedefs in types.h escape into the wider world and conflict

### ISSUE THREAD

**Issue by Kazade** [KallistiOS/KallistiOS#840]
If you try to compile stb_vorbis with KOS, you get the following errors (snipped)

```
stb_vorbis.c:656:21: error: conflicting types for 'int8'; have 'signed char'
  656 | typedef signed char int8;
      |                     ^~~~
In file included from /opt/toolchains/dc/kos/include/sys/_types.h:206,
                 from /opt/toolchains/dc/sh-elf/sh-elf/include/sys/reent.h:16,
                 from /opt/toolchains/dc/sh-elf/sh-elf/include/stdio.h:60,
                 from stb_vorbis.c:84:
/opt/toolchains/dc/kos/kernel/arch/dreamcast/include/arch/types.h:39:14: note: previous declaration of 'int8' with type 'int8' {aka 'char'}
   39 | typedef char int8;                  /**< \brief 8-bit signed integer */
      |              ^~~~
stb_vorbis.c:659:22: error: conflicting types for 'uint32'; have 'unsigned int'
  659 | typedef unsigned int uint32;
```

This is because stb_vorbis has the following typedefs: 

```
typedef unsigned char uint8;
typedef signed char int8;
typedef unsigned short uint16;
typedef signed short int16;
typedef unsigned int uint32;
typedef signed int int32;
```

I haven't found an easy fix to this, I have previously ifdef'd out those typedefs, but in GCC 14 I'm now seeing errors due to pointer mismatches (where stb_vorbis uses unsigned int instead of uint32). 

I don't think KOS should be exposing these typedefs through stdio.h? I think that's the issue...

**Comment by gyrovorbis** [comment]
Yes, it's quite a PITA. This is the same issue we've had with Sonic Mania and a few other codebases too. I'll see if I can help look into this as well, because it's also been a total annoyance to me too.

Basically in Mania, I believe we had to rearrange include files and do a bunch of shenanigans... but you don't always own the C/C++ files to do that, either... :(

**Comment by QuzarDC** [comment]
This is caused by this: https://github.com/KallistiOS/KallistiOS/blob/99308c534d03c585d4b4aa14ad278e072081300b/include/sys/_types.h#L201-L202

Looks like it was added intentionally in '04 during the initial switch/cleanup to newlib in order to preserve the old old KOS behavior. I'd be in favor of removing it, but it would likely force random DC software to now pull in some kos header in every file if they were blindly using those types because they *just worked* (or perhaps because they were working off examples and we provided them there). It *is* also pulled in via `kos.h` so it would truly only be for something like an internal header where a person is using the types and only including `stdio` not realizing that they're using custom types.

I took a sec to try rebuilding KOS with it removed and we for sure have some places that rely inappropriately on getting our custom types from stdio rather than `<arch/types>` (or using standard types). That would have to, at least, be resolved throughout the codebase (and kos-ports) as a first step.

This perhaps should be paired with a proper effort to move off these custom types finally. I agreed with the original approach of only replacing them with standard types when making other changes, but we should eventually take a leap to close the rest of the gap.

**Comment by QuzarDC** [comment]
Moved this up to v2.3.0 as substantial work was done to reduce it's impact so far that will be in v2.2.0. I'll aim to remove that `#include <arch/types.h>` after the v2.2.0 release so that there's the most time until a next stable to address issues it might cause.

Additionally, I think we might be close enough to finally finish up dropping them entirely in favor of stdint and will see about moving that along in a reviewable way.
